Skeleton for Dashboard & Navbar

ruslan-akhm opened the pull request:
- Added skeleton for dashboard  (accessible via "/dashboard" in URL, authentication hardcoded by now ) + Mobile layout
- Added Navbar styling for authenticated user (style only, no options yet)
- Minor fixes
- 
UPD: Based on Thomas's code in branch "IN-Authentication-UserContext-ProtectedRoutes" refactored Protected Routes and Context. Plus added Logout button to dashboard to later connect it with BE 

![1](https://user-images.githubusercontent.com/47538460/104514974-20f5e600-55c0-11eb-8b76-ee22695833dd.jpg)
![2](https://user-images.githubusercontent.com/47538460/104515020-33701f80-55c0-11eb-891f-84ced69da06f.jpg)
![3](https://user-images.githubusercontent.com/47538460/104515025-366b1000-55c0-11eb-83bb-8f7a89c30834.jpg)
![4](https://user-images.githubusercontent.com/47538460/104515033-38cd6a00-55c0-11eb-85a9-2f05b02d879e.jpg)

ruslan-akhm committed:
Skeleton for Dashboard & Navbar

ruslan-akhm committed:
Added Protected Routes and Fixed Context

dproc96 commented on client/src/App.js:
```diff
@@ -1,28 +1,34 @@
-        <Route path="/">
-          {loggedIn ? <Profile /> : <Redirect to="/signup" />}
+        <Navbar />
+        <Route path="/" exact>
```
You could save some code by using a ProtectedRoute here to store the Redirect

ruslan-akhm committed:
Updated package.json with MUI icons

dproc96 commented on client/src/components/LoginForm.js:
```diff
@@ -61,27 +61,31 @@ function LoginForm() {
       </Grid>
       <Grid item className={classes.elements}>
-        <Typography>Email</Typography>
+        {/*<Typography>Email</Typography>*/}
```
It would be better to remove these instead of commenting them out

dproc96 commented on client/src/components/navbar/Navbar.js:
```diff
@@ -0,0 +1,90 @@
+  toolbarAuth: {
+    backgroundColor: "white",
+  },
+  sitterLink: { color: "black", marginRight: "30px" },
```
You should be able to get rid of the underline for this by adding textDecoration: "none" to closer match the mocks

ruslan-akhm replied:
But this one is according to mocks isnt it?

dproc96 replied:
Oops, you're right! That's my bad

dproc96 commented on client/src/components/navbar/mobileNavbar.js:
```diff
@@ -0,0 +1,112 @@
+    height: "80px",
+    margin: "30px auto 30px",
+  },
+  menuButton: {
```
You'll probably want to get rid of the underline for these as well

ruslan-akhm replied:
Okay. I just personally thought It looked better with them :)

dproc96 replied:
Ah okay, feel free to stick with it then since there's no mocks for the mobile view. I wasn't sure if it was intentional or not

dproc96 commented on client/src/components/navbar/Navbar.js:
```diff
@@ -0,0 +1,90 @@
+            </Link>
+          </Box>
+          <Link to="/signup" style={{ textDecoration: "none" }}>
+            <Button color="secondary" variant="contained" size="large">
```
The color for this button doesn't seem to match the rest of the app. Can you check that? May need to be color primary

ruslan-akhm replied:
But this one is according to mocks as well. Are my mocks outdated?

dproc96 replied:
The color in the mocks matches the logo (#f04040) according to my eyedropper tool

ruslan-akhm replied:
I can see it now too. Barely noticable with naked eye

dproc96 replied:
[This](https://chrome.google.com/webstore/detail/eye-dropper/hmdcmlfkchdmnmnmheododdhjedfccka/related?hl=en) is a good tool for checking colors match in cases like this

dproc96 commented on client/src/context/Context.js:
```diff
@@ -0,0 +1,24 @@
+
+export const UserContext = createContext();
+
+export default ({ children }) => {
```
You should assign an arrow function to a variable before exporting it

dproc96 commented on client/src/pages/Dashboard.js:
```diff
@@ -0,0 +1,165 @@
+
+  //rendering windows corresponding to menu options clicked
+  function renderSwitch(pageShown) {
+    switch (pageShown) {
```
Would it make more sense to use Routes and Links instead of a Switch statement? Something like "/profile/{subpage}"

ruslan-akhm replied:
It depends on if we want the URL change at every option or not. We might want it changed though since we would make post request at every "page". I'll take a look at that

dproc96 reviewed (commented):
Overall the UI looks really good! Just noted a few changes here. Also I noticed a few warnings in the console when I ran it locally, can you address those as well? For the classes variables, I think it would be better to use them rather than delete them in order to address the warning.

ruslan-akhm commented:
> Overall the UI looks really good! Just noted a few changes here. Also I noticed a few warnings in the console when I ran it locally, can you address those as well? For the classes variables, I think it would be better to use them rather than delete them in order to address the warning.

Classes warning are coming from components that are just "dummy data" for now as it's yet a skeleton, but I would comment them out by now. Same for context variables isAuth/setIsAuth which are not in use yet since we didn't integrate FE and BE together, but already are called by components that will use them (mobileNavbar), but I'll delete them too.

ruslan-akhm committed:
Updated routing in dashboard skeleton

Routing and Protected routes updated
Errors removed
Minor styling fixes implemented

ruslan-akhm commented:
Routing and Protected routes updated
Errors removed
Minor styling fixes implemented

dproc96 reviewed (approved):
Looks good to go!

ruslan-akhm committed:
Merge branch 'dev' into dashboard-skeleton-and-navbar
